invenio_app_ils/facets.py

```diff
@@ -48,16 +48,16 @@ def inner(values):
     return inner
 
 
-def not_empty_object_or_list_filter(field):
-    """Create a custom exists filter.
+def eitems_has_files_filter(field):
+    """Create a custom filter for eitems with files.
 
     :param field: Field name.
-    :missing_val
-    :returns: Function that returns the Terms query.
     """
     def inner(values):
-        return Bool(**{'must': {'exists': {'field': field}}})
-
+        if 'has_files' in values:
+            return Bool(**{'must': {'exists': {'field': field}}})
+        else:
+            return Bool(**{'must_not': {'exists': {'field': field}}})
```

Comment:
we have the access to the field name from the outer function, so instead of making it specialised to only e-item, we can do
```suggestion
        if field in values:
            return Bool(**{'must': {'exists': {'field': field}}})
        else:
            return Bool(**{'must_not': {'exists': {'field': field}}})
```
and it stays as a generic empty/list filter.

Reply:
Yes! You are right, we can keep it as a generic filter, but we need to pass the filter value as an argument (since field value is `files.file_id`, see [here](https://github.com/inveniosoftware/invenio-app-ils/pull/1006/files#diff-5cb3b7e47a3af9c319af5e812e25d1297eab7f43aa4c9671be71a481b3fa8d9bR865))